### PR TITLE
Add pre-commit hook and setup script

### PR DESCRIPTION
**Pull request by skddc**
Runs the appropriate linter on staged files before committing.

Can be easily installed in the local repo by running `npm run setup`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
     "lint:contracts": "solhint \"contracts/**/*.sol\" \"apps/*/contracts/**/*.sol\"",
     "lint:contract-tests": "eslint apps/*/test",
     "lint:wrapper": "eslint lib/",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "setup": "sh scripts/git-hooks/install"
```
Comment (bumi) [inline package.json]:
```suggestion
    "setup-git-hooks": "sh scripts/git-hooks/install"
```

I find this name a bit confusing as it sounds it has something todo with the setup to install/run the code. We have `bootstrap`, `reset`, `install-all` - so I'd by a bit more specific here as so far this setup task is only installing the git hooks.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,11 @@ Each of the aragon apps are separate packages:
 
 You can use `npm run install-all` to install all app dependencies at once.
 
+We also recommend using our git hooks for linting staged files and other
+convenient helpers. You can install them via:
+
+    $ npm run setup
```
Comment (bumi) [inline README.md]:
```suggestion
    $ npm run setup-git-hooks
```

File: package.json (hunk 1 above)
Comment (skddc) [reply package.json]: This setup is meant to include anything that is necessary, other than `npm install` to set up the repo itself.

Comment (skddc) [reply package.json]: Moved the entire thing to `postinstall` now.

**Comment by skddc** [comment]
There's a weird issue with npm, in that when running a `postinstall` script it re-validates the entire npm dependency stack. I don't know why that is.

**Comment by skddc** [comment]
@bumi Can we merge this now?

**Comment by bumi** [comment]
Isn't `postinstall` also executed when this is installed as a dependency? e.g. in kredits-web? 

if it only runs when running `npm install` I am ok with merging it. though as much as I need this am not a fan of postinstall scripts doing such modifications. 
when we tried to run the app installations in a post-install script it was also only causing problems (don't remember the details though).

**Comment by skddc** [comment]
Renamed it to your suggestion.

**Pull request merged**